dropwizard-servlets/src/main/java/io/dropwizard/servlets/tasks/LogConfigurationTask.java

```diff
@@ -57,10 +62,28 @@ public LogConfigurationTask(ILoggerFactory loggerContext) {
     public void execute(ImmutableMultimap<String, String> parameters, PrintWriter output) throws Exception {
         final List<String> loggerNames = getLoggerNames(parameters);
         final Level loggerLevel = getLoggerLevel(parameters);
+        final Duration duration = getDuration(parameters);
 
         for (String loggerName : loggerNames) {
-            ((LoggerContext) loggerContext).getLogger(loggerName).setLevel(loggerLevel);
-            output.println(String.format("Configured logging level for %s to %s", loggerName, loggerLevel));
+            Logger logger = ((LoggerContext) loggerContext).getLogger(loggerName);
+
+            String message = String.format("Configured logging level for %s to %s", loggerName, loggerLevel);
+
+            if (loggerLevel != null && duration != null) {
+                // Get the effective level before setting the new level
+                final Level effectiveLevel = logger.getEffectiveLevel();
+                final long millis = duration.toMillis();
+                timer.schedule(new TimerTask() {
```

Comment:
What would happen if a user wants to override a previously set timeout for a logger or remove the timeout?

Currently, the logger would be reset to the initial log level by the first scheduled task anyway and the second scheduled task would restore the previously set log level.

Example:

* Initial log level is "WARN".
* User sends logger level "DEBUG" and duration P10m.
* -- 8 minutes pass --
* User sends logger level "DEBUG" and duration P30m.
* -- 2 minutes pass --
* Scheduled task resets log level to "WARN"
* -- 28 minutes pass --
* Scheduled task resets log level to "DEBUG"
* ???
* PROFIT!

Reply:
> Currently, the logger would be reset to the initial log level by the first scheduled task anyway and the second scheduled task would restore the previously set log level.

I went back and forth on this. Should it be restored to the previous effective level or `null`? When `null` is supplied logback will set to the [effective parent level](https://github.com/qos-ch/logback/blob/4dd08e68321548cd85251ec389f5a832505e1268/logback-classic/src/main/java/ch/qos/logback/classic/Logger.java#L157-L158). This would eliminate some of the ordering issues, as 

>What would happen if a user wants to override a previously set timeout for a logger or remove the timeout?

I had a hunch this would come up. I had originally gone down that path and it was getting a little more complex than I wanted to introduce. My second pass implementation was to keep it really simple, with some of those caveats on ordering of operations.

We could push the lifecycle from the environment down to the admin environment.
https://github.com/dropwizard/dropwizard/blob/2cfdd6dc5bd390e3f5b1ef47d8f7e7fd9efd7fd9/dropwizard-core/src/main/java/io/dropwizard/setup/Environment.java#L76-L78

Use that to create a scheduled executor service for the log configuration task
https://github.com/dropwizard/dropwizard/blob/2cfdd6dc5bd390e3f5b1ef47d8f7e7fd9efd7fd9/dropwizard-lifecycle/src/main/java/io/dropwizard/lifecycle/setup/LifecycleEnvironment.java#L70

Then use weak concurrent hash map to store package/future mappings, so that we can support `doDelete` on the task servlet to support removal. Could also enforce that there is only one "active" log level change per package at a time, or when a new one comes it replaces the existing override.
https://github.com/dropwizard/dropwizard/blob/2cfdd6dc5bd390e3f5b1ef47d8f7e7fd9efd7fd9/dropwizard-servlets/src/main/java/io/dropwizard/servlets/tasks/TaskServlet.java#L96

Reply:
I haven't looked too close at the impl but I'm in agreement that when believe the situation where the initial level is `WARN` and there are two tasks to set the level to `DEBUG` and `INFO` then no matter when the two tasks are executed or finished the final log level should be set back to `WARN`